### PR TITLE
Added GSoC stats in home page

### PR DESCRIPTION
**Pull request by aritroCoder**
Added graphs for different GSOC stats and also a side pane which contains different data related to all previous year GSOC

### CODE REVIEW

**Comment by Sparsh1212** [comment]
Can you also attach the Demo video

**Comment by aritroCoder** [comment]
https://user-images.githubusercontent.com/92646038/145704450-77a447ac-2184-471a-aece-5702d9e2868b.mp4

**Review by Sparsh1212 — CHANGES_REQUESTED** [review]
Some high-level comments:
- The info appearing on the right looks like a repetition of the graphs on the left. It will be best to remove the textual info as all of that could be interpreted from the graphs.
- Since, this is the launcher page, it's fine to render the overall stats of GSoC. By that I mean to say, there's no need for a "Show Summary" button. Please remove that.
- Since we want to make the search distraction-free, so please remove the show summary button from the search too. It will be best to remove it completely from the code.
- Regarding the UI. Finally, I think, it will be best to only render the graphs. That being said, make sure you center the new graphs added. In the first row, center the twp graphs and increase their size (height and width) so that they take the max-width. In the second row, just center the organization's graph, and the size should be around 60-70% of the width.
- The UI is not mobile responsive. In the mobile view, it will be best if each graph can take the full width and we can render the in the form of columns.

**Comment by aritroCoder** [comment]
> Some high-level comments:
> 
> * The info appearing on the right looks like a repetition of the graphs on the left. It will be best to remove the textual info as all of that could be interpreted from the graphs.
> * Since, this is the launcher page, it's fine to render the overall stats of GSoC. By that I mean to say, there's no need for a "Show Summary" button. Please remove that.
> * Since we want to make the search distraction-free, so please remove the show summary button from the search too. It will be best to remove it completely from the code.
> * Regarding the UI. Finally, I think, it will be best to only render the graphs. That being said, make sure you center the new graphs added. In the first row, center the twp graphs and increase their size (height and width) so that they take the max-width. In the second row, just center the organization's graph, and the size should be around 60-70% of the width.
> * The UI is not mobile responsive. In the mobile view, it will be best if each graph can take the full width and we can render the in the form of columns.

Ok , thanks for your detailed review. I am working on your points

**Comment by aritroCoder** [comment]
I have made the required changes. Please check the video. If everything looks fine I will commit the code and re initiate the PR.

https://user-images.githubusercontent.com/92646038/145723638-11410f45-2c36-4ac5-9a8c-db3e65da97de.mp4

**Comment by Sparsh1212** [comment]
> I have made the required changes. Please check the video. If everything looks fine I will commit the code and re initiate the PR.
> 
>  2021-12-12.23-21-03.mp4

Good job on the suggestions.
After seeing the demo video, I realized that if we individually display the graphs in 100% width even on the desktop view, they look awesome. So final UI changes will be:
- Make each graph almost 90% width and increase the height a bit. So in total there will be a total of 3 rows in both Desktop and Mobile View.
- Increase the spacing between the rows of graphs.
- I was thinking to remove all the launcher data content and keep it a two liner. That being said, remove all the existing content present which starts from "A blazingly fast tool to .... bookmark your favourite organization" and replace it with these two lines: 
1. "A blazingly-fast tool to analyze all the organizations selected in Google Summer of Code."
2. "Start analyzing now!"

**Comment by Sparsh1212** [comment]
@aritroCoder Any updates on this?

**Comment by aritroCoder** [comment]
> > I have made the required changes. Please check the video. If everything looks fine I will commit the code and re initiate the PR.
> > 2021-12-12.23-21-03.mp4
> 
> Good job on the suggestions. After seeing the demo video, I realized that if we individually display the graphs in 100% width even on the desktop view, they look awesome. So final UI changes will be:
> 
> * Make each graph almost 90% width and increase the height a bit. So in total there will be a total of 3 rows in both Desktop and Mobile View.
> * Increase the spacing between the rows of graphs.
> * I was thinking to remove all the launcher data content and keep it a two liner. That being said, remove all the existing content present which starts from "A blazingly fast tool to .... bookmark your favourite organization" and replace it with these two lines:
> 
> 1. "A blazingly-fast tool to analyze all the organizations selected in Google Summer of Code."
> 2. "Start analyzing now!"

Yes I am working on it, it will be ready by tommorow. Thank you!

**Comment by aritroCoder** [comment]
https://user-images.githubusercontent.com/92646038/145856287-4aa59c21-123e-469c-9f79-3bd651d9496e.mp4

I have made the requested changes, please see them and give any feedback

**Comment by Sparsh1212** [comment]
> 2021-12-13.22-33-56.mp4 
> I have made the requested changes, please see them and give any feedback

The UI looks excellent. 
I though observed one small problem with the graph caption text on the mobile screen. Observe the graph caption at the bottom of the graph, please decrease its font size for mobile view. Rest the UI looks perfect to me. Push your code after this suggested fix and then we'll have a final round of code review and then get set go!

**Comment by aritroCoder** [comment]
> > 2021-12-13.22-33-56.mp4
> > I have made the requested changes, please see them and give any feedback
> 
> The UI looks excellent. I though observed one small problem with the graph caption text on the mobile screen. Observe the graph caption at the bottom of the graph, please decrease its font size for mobile view. Rest the UI looks perfect to me. Push your code after this suggested fix and then we'll have a final round of code review and then get set go!

I came at a different kind of solution for this. React chartjs does not use any CSS, as it renders the chart as a image. So I have used a logic at the code where it will decide the chart font size at the time of loading the screen . As the users wont be changing their screen size, so I think it will not be a issue. Do give your feedback

https://user-images.githubusercontent.com/92646038/145866500-d264c654-17c8-4b53-8d71-ddbf640618d3.mp4

**Comment by Sparsh1212** [comment]
> > > 2021-12-13.22-33-56.mp4
> > > I have made the requested changes, please see them and give any feedback
> > 
> > 
> > The UI looks excellent. I though observed one small problem with the graph caption text on the mobile screen. Observe the graph caption at the bottom of the graph, please decrease its font size for mobile view. Rest the UI looks perfect to me. Push your code after this suggested fix and then we'll have a final round of code review and then get set go!
> 
> I came at a different kind of solution for this. React chartjs does not use any CSS, as it renders the chart as a image. So I have used a logic at the code where it will decide the chart font size at the time of loading the screen . As the users wont be changing their screen size, so I think it will not be a issue. Do give your feedback
> 
>  2021-12-13.23-46-18.mp4

The approach sounds so cool to me. 
Also, always don't hesitate to push your code whenever you make a change to your PR. Viewing the change code also helps the reviewer to guess whether the solution has been implemented in the correct way or not. Otherwise, it will lead to unnecessary extra iterations.  :)

**Comment by aritroCoder** [comment]
I have committed the changes to the PR

**Comment by aritroCoder** [comment]
Did you check the code? Please check and update your status. Thank you

**Review by Sparsh1212 — CHANGES_REQUESTED** [review]
- Please remove the commented code
- Make all the three graphs, part of the `LaunchingComponent` . This is because they are a part of the launching screen only, also upon changing the state `display launcher`, you would not have to programmatically make its display to be none. They will automatically be gone from the screen as the LaunchingComponent gets disappeared.

File: src/components/CompanyChart.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                            data: [42,102,135,175,150,150,175,180,177,190,134, 178, 198, 206, 201, 199, 199],
+                            backgroundColor: 'rgba(54, 162, 235, 0.2)',
+                            // [
+                            //     'rgba(255, 99, 132, 0.2)',
+                            //     'rgba(54, 162, 235, 0.2)',
+                            //     'rgba(255, 206, 86, 0.2)',
+                            //     'rgba(75, 192, 192, 0.2)',
+                            //     'rgba(153, 102, 255, 0.2)',
+                            //     'rgba(255, 159, 64, 0.2)',
+                            // ],
```
Comment (Sparsh1212) [inline src/components/CompanyChart.js:25]: Remove.

File: src/components/CompanyChart.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                            borderColor: 'rgba(54, 162, 235, 1)',
+                            // [
+                            //     'rgba(255, 99, 132, 1)',
+                            //     'rgba(54, 162, 235, 1)',
+                            //     'rgba(255, 206, 86, 1)',
+                            //     'rgba(75, 192, 192, 1)',
+                            //     'rgba(153, 102, 255, 1)',
+                            //     'rgba(255, 159, 64, 1)',
+                            // ],
```
Comment (Sparsh1212) [inline src/components/CompanyChart.js:34]: Remove.

File: src/components/CompanyChart.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                            borderWidth: 1,
+                        },
+                    ],
+                    }
+                }
+
+                // height="200px"
+                // width="300px"
```
Comment (Sparsh1212) [inline src/components/CompanyChart.js:42]: Remove.

File: src/components/GraphChart.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+                            data: [410,630,905,1126,1000,1026,1115,1212,1192,1307,1051, 1206, 1318, 1264, 1276, 1198, 1292],
+                            backgroundColor: 'rgba(255, 99, 132, 0.2)',
+                            // [
+                            //     'rgba(255, 99, 132, 0.2)',
+                            //     'rgba(54, 162, 235, 0.2)',
+                            //     'rgba(255, 206, 86, 0.2)',
+                            //     'rgba(75, 192, 192, 0.2)',
+                            //     'rgba(153, 102, 255, 0.2)',
+                            //     'rgba(255, 159, 64, 0.2)',
+                            // ],
```
Comment (Sparsh1212) [inline src/components/GraphChart.js:26]: Remove.

File: src/components/GraphChart.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+                            borderColor: 'rgba(255, 99, 132, 1)',
+                            // [
+                            //     'rgba(255, 99, 132, 1)',
+                            //     'rgba(54, 162, 235, 1)',
+                            //     'rgba(255, 206, 86, 1)',
+                            //     'rgba(75, 192, 192, 1)',
+                            //     'rgba(153, 102, 255, 1)',
+                            //     'rgba(255, 159, 64, 1)',
+                            // ],
```
Comment (Sparsh1212) [inline src/components/GraphChart.js:35]: Remove.

File: src/components/GraphChart.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+                            backgroundColor: 'orange',
+                            borderColor: 'red',
+                            borderWidth: 1,
+                        },
+                    ],
+                    }
+                }
+
+                // height="200px"
+                // width="300px"
```
Comment (Sparsh1212) [inline src/components/GraphChart.js:50]: Remove.

File: src/components/Home.js [hunk 7]
```diff
@@ -7,26 +7,43 @@ import AdvancedSearch from './AdvancedSearch';
...
 
-const descendingSortByYear = (resultList) =>{
-  return resultList.sort( (a,b) => { 
-      return (b.year.length - a.year.length)
-    });
+
+const descendingSortByYear = (resultList) => {
+  return resultList.sort((a, b) => {
+    return (b.year.length - a.year.length)
+  });
```
Comment (Sparsh1212) [inline src/components/Home.js:20]: Revert the formatting change. Ditto for overall PR. We don't want the formatting to be changed.

File: src/components/Home.js [hunk 8]
```diff
@@ -64,16 +81,26 @@ const Home = ({bookmarked, setBookmarked}) => {
 
   const AdvancedSearchRef = useRef();
 
+
```
Comment (Sparsh1212) [inline src/components/Home.js:84]: Remove.

**Comment by Sparsh1212** [comment]
Also, please rename the Chart file names with more descriptive names. Replace `Company` with `Organization` everywhere in the naming. Also, if it's easy for you then please try to group all the three graph files into a separate folder under `components` named as `launcher`. And there put all the files of `LaunchingComponent`. This ensures good modularity of the code.

**Comment by aritroCoder** [comment]
I have done all your requested changes. Please go through the code now and give your suggestions

File: src/components/Home.js [hunk 9]
```diff
@@ -7,15 +7,17 @@ import AdvancedSearch from './AdvancedSearch';
 import Footer from './Footer';
 import { VerticleButton as ScrollUpButton } from 'react-scroll-up-button';
 import LaunchingComponent from './LaunchingComponent';
-import { Link } from 'react-router-dom'
+import { Link } from 'react-router-dom';
+
```
Comment (Sparsh1212) [inline src/components/Home.js:11]: Revert this change

File: src/components/Home.js [hunk 10]
```diff
@@ -7,15 +7,17 @@ import AdvancedSearch from './AdvancedSearch';
...
 
 const descendingSortByYear = (resultList) =>{
-  return resultList.sort( (a,b) => { 
-      return (b.year.length - a.year.length)
-    });
+  return resultList.sort( (a,b) => {
+    return (b.year.length - a.year.length)
+  });
```
Comment (Sparsh1212) [inline src/components/Home.js:16]: Revert the formatting change.

File: src/components/Home.js [hunk 11]
```diff
@@ -7,15 +7,17 @@ import AdvancedSearch from './AdvancedSearch';
...
 }
 
 const Home = ({bookmarked, setBookmarked}) => {
+
```
Comment (Sparsh1212) [inline src/components/Home.js:20]: Remove blank line.

File: src/components/Home.js [hunk 12]
```diff
@@ -27,6 +29,7 @@ const Home = ({bookmarked, setBookmarked}) => {
 
   const buildSearchList = (search, filter) => {
     setDisplayLauncher(false);
+
```
Comment (Sparsh1212) [inline src/components/Home.js:32]: Remove blank line.

File: src/components/Home.js [hunk 13]
```diff
@@ -39,7 +42,7 @@ const Home = ({bookmarked, setBookmarked}) => {
           return false;
         })
       );
-      
+
```
Comment (Sparsh1212) [inline src/components/Home.js:45]: Revert this change.

File: src/components/Home.js [hunk 14]
```diff
@@ -72,8 +75,12 @@ const Home = ({bookmarked, setBookmarked}) => {
         </Header>
         <Link to="/bookmarks" className="nav-button">Bookmarks</Link>
 
+
         <AdvancedSearch ref={AdvancedSearchRef} buildSearchList={buildSearchList} />
         {displayLauncher && <LaunchingComponent />}
+
+
+
```
Comment (Sparsh1212) [inline src/components/Home.js:83]: Remove these blank lines.

**Review by Sparsh1212 — CHANGES_REQUESTED** [review]
Almost LGTM!
Just some nit suggestions added. PTAL.

**Comment by Sparsh1212** [comment]
Also, please take care of the merge conflict.

**Comment by aritroCoder** [comment]
Done :)

File: src/components/Home.js [hunk 15]
```diff
@@ -9,10 +9,10 @@ import { VerticleButton as ScrollUpButton } from 'react-scroll-up-button';
 import LaunchingComponent from './LaunchingComponent';
 import { Link } from 'react-router-dom'
 
-const descendingSortByYear = (resultList) =>{
+const descendingSortByYear = (resultList) =>{ 
   return resultList.sort( (a,b) => { 
-      return (b.year.length - a.year.length)
-    });
+    return (b.year.length - a.year.length)
+   });
```
Comment (Sparsh1212) [inline src/components/Home.js:15]: I think this got missed :)

File: src/components/Home.js [hunk 16]
```diff
@@ -77,7 +77,6 @@ const Home = ({bookmarked, setBookmarked}) => {
           <h1 id = 'mainHeader' onClick={reRenderLauncher}> GSoC Analyser </h1>
         </Header>
         <Link to="/bookmarks" className="nav-button">Bookmarks</Link>
-
```
Comment (Sparsh1212) [inline src/components/Home.js:80]: Revert

File: src/components/LaunchingComponent.js [hunk 17]
```diff
@@ -3,23 +3,30 @@ import React from 'react';
 import Typist from 'react-typist';
 import '../css/mainpagecss.css';
 
+//Importing the charts files
```
Comment (Sparsh1212) [inline src/components/LaunchingComponent.js:6]: We don't need this comment

File: src/components/LaunchingComponent.js [hunk 18]
```diff
@@ -3,23 +3,30 @@ import React from 'react';
...
-            organizations selected in GSoC.
+          A blazingly-fast tool to analyze all the organizations selected in Google Summer of Code.
           </p>
-          <Typist.Delay ms={150} />
-          <p>Search by tech-stack / organisation name / category or by topic.</p>
           <Typist.Delay ms={200} />
-          <p>Start Searching now! Use comma to separate multiple tags in tech or topics.</p>
-          <Typist.Delay ms={250} />
-          <p>Also Bookmark your favorite organisation.</p>
+          <p>Start Analyzing now!</p>
```
Comment (Sparsh1212) [inline src/components/LaunchingComponent.js:20]: Replace this with: `Start Analyzing now! (Use a comma to separate multiple tags in tech/topic)`

File: src/css/mainpagecss.css [hunk 19]
```diff
@@ -5,6 +5,17 @@
   color: white;
   width: 100%;
 }
+/* 
+#gsocStat{
+
+} */
```
Comment (Sparsh1212) [inline src/css/mainpagecss.css:11]: Redundant code.

**Review by Sparsh1212 — CHANGES_REQUESTED** [review]
Some more nits :)
Also can you please add a final video proof after making all these changes?

**Comment by aritroCoder** [comment]
If this commit is perfect, I will create the final video

**Comment by Sparsh1212** [comment]
> If this commit is perfect, I will create the final video

Yeah! things look pretty good to me in the latest commit. Just upload a final video and if everything looks good, then we can go for a merge :)

**Comment by aritroCoder** [comment]
https://user-images.githubusercontent.com/92646038/146202634-f89a1a0c-424b-40ff-81c1-ad482445ec4f.mp4

https://user-images.githubusercontent.com/92646038/146202736-d32c9ec9-d364-4bf7-a588-1eeb15f58cb7.mp4

second video is for mobile view. 
I think the gap between the typewriter text and chart is too much if you want I can reduce that too

**Comment by Sparsh1212** [comment]
> 2021-12-15.19-44-23.mp4 
>  2021-12-15.19-45-00.mp4 
> second video is for mobile view. I think the gap between the typewriter text and chart is too much if you want I can reduce that too

Yes, that will be really good if you could reduce the gap. You can achieve that simply by decreasing the `min-height` of the animatedDiv I think. Also, it will be good if you can slightly increase the separation between the graphs. Don't increase it too much, just a bit.

**Comment by Sparsh1212** [comment]
I think after these two fixes, we are fully set to go for a merge.

**Comment by Sparsh1212** [comment]
> > 2021-12-15.19-44-23.mp4
> > 2021-12-15.19-45-00.mp4
> > second video is for mobile view. I think the gap between the typewriter text and chart is too much if you want I can reduce that too
> 
> Yes, that will be really good if you could reduce the gap. You can achieve that simply by decreasing the `min-height` of the animatedDiv I think. Make sure to decrease it for the mobile view as well. Also, it will be good if you can slightly increase the separation between the graphs. Don't increase it too much, just a bit.

**Comment by aritroCoder** [comment]
https://user-images.githubusercontent.com/92646038/146206131-0b483f5f-6b10-4081-81d9-ef034adad889.mp4

https://user-images.githubusercontent.com/92646038/146206193-c3e00041-ae86-405b-bd0d-acd321716c4a.mp4

Done !

**Comment by Sparsh1212** [comment]
Just one last change.
You don't need to reupload the video again after this.
Make the min-height in the PC view to be 34vh instead of 54vh. That's it.

**Comment by aritroCoder** [comment]
Alright, I did that :)

**Review by Sparsh1212 — APPROVED** [review]
LGTM!
Must say, commendable job :)

**Pull request merged**